### PR TITLE
Twitch.tv support

### PR DESCRIPTION
**Pull request by dvtkrlbs**
It is still work in progress i am openning it to brainstorm

### CODE REVIEW

File: rsget_lib/src/plugins/twitch.rs [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        let username_re = Regex::new(r"^(?:https?://)?(?:www\.)?twitch\.tv/([a-zA-Z0-9]+)")?;
+        let cap = match username_re.captures(&url) {
+            Some(capture) => capture,
+            None => return Err(StreamError::Rsget(RsgetError::new("Cannot capture usernane"))),
+        };
+
+        // It  Error running: Reqwest(Error { kind: Url(RelativeUrlWithoutBase), url: None })
+        let client_id = match env::var("TWITCH_TOKEN") {
+            Ok(val) => val,
+            Err(_) => return Err(StreamError::Rsget(RsgetError::new("No capture found"))),
```
Comment (Erk-) [inline rsget_lib/src/plugins/twitch.rs]: Return a better error here maybe something like: `Twich token not set`

File: rsget_lib/src/plugins/mod.rs [hunk 2]
```diff
@@ -5,3 +5,4 @@ pub mod douyin;
 pub mod tiktok;
 pub mod huya;
 pub mod dlive;
+pub mod twitch;
```
Comment (Erk-) [inline rsget_lib/src/plugins/mod.rs]: add newline in the end of the file

File: rsget_lib/src/plugins/twitch.rs [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    fn download(&self, path: String) -> Result<u64, StreamError> {
+        if !self.is_online() {
+            Err(StreamError::Rsget(RsgetError::new("Stream offline")))
+        } else {
+            let file = File::create(path)?;
+            let stream = Stream::new(self.get_stream()?);
+            Ok(stream.write_file(&self.client.rclient, file)?)
+        }
+    }
+}
```
Comment (Erk-) [inline rsget_lib/src/plugins/twitch.rs]: also a newline here

**Comment by dvtkrlbs** [comment]
did all but there is 2 small problems i think download function is not working correctly (--play works fine) and in the env part no matter what i return as an error the error message becomes `Reqwest(Error { kind: Url(RelativeUrlWithoutBase), url: None })`

**Pull request merged**